### PR TITLE
Refs #9848: Move config file to config directory and provide via gem.

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by ehelms** [comment]
Identified this while adding this to katello-packaging and noted that this is the way other hammer modules handle it.

File: rubygem-hammer_cli_katello.spec [hunk 1]
```diff
@@ -52,7 +51,7 @@ gem install --local --install-dir .%{gem_dir} \
 
 %install
 mkdir -p %{buildroot}%{_sysconfdir}/%{confdir}/cli.modules.d
-install -m 755 %{SOURCE1} %{buildroot}%{_sysconfdir}/%{confdir}/cli.modules.d/katello.yml
+install -m 755 .%{gem_instdir}/config/katello.yml %{buildroot}%{_sysconfdir}/%{confdir}/cli.modules.d/katello.yml
```
Comment (jlsherrill) [inline rubygem-hammer_cli_katello.spec]: i can't actually build the rpm with this change.  For one i think its geminstdir

but even that doesn't get you all the way

Comment (jlsherrill) [inline rubygem-hammer_cli_katello.spec]: needs a %file entry

Comment (jlsherrill) [inline rubygem-hammer_cli_katello.spec]: update to use  gem_instdir and gem_name throughout

**Comment by jlsherrill** [comment]
ACK

**Pull request merged**